### DISCUSSION TITLE
Build times much longer: ComposableArchitectureMacros

### DISCUSSION THREAD

**Discussion by etietz** [pointfreeco/swift-composable-architecture#2776] [Q&A] [answered]
I've successfully upgraded a project from TCA 0.57 to 1.7.*, and everything is working well.  However, my build times are ~8x longer (than on TCA 0.57), and during the build, all my CPU cores are over 95% with a process named ComposableArchitectureMacros.  Can anyone tell me if I have Xcode configured incorrectly?  I'm using Xcode 15.2 (15C500b) on a M3 Max (16 cores) with 128GB of memory.

**Comment by jshier** [comment, marked as answer]
If you're building in release mode, that's just swift-syntax. Otherwise it may just be a build issue. Nothing to configure, but there may be some local workarounds you can do (https://forums.swift.org/t/compilation-extremely-slow-since-macros-adoption/67921/59) but it's really an Apple issue.

**Reply by etietz** [reply]
Thanks!

**Comment by mbrandonw** [comment]
Yeah, this is a well known problem with macros, and so far Apple has not provided any guidance on how to solve it. The problem is that using macros largely requires SwiftSyntax, which currently takes 15-20 seconds to compile in debug and 2+ minutes in release.

**Reply by Narsail** [reply]
Any plans on pointing the library to a workaround like https://github.com/johnno1962/InstantSyntax?

**Reply by mbrandonw** [reply]
Hi @Narsail, there is nothing we need to do integrate this library into ours. If you follow the instructions you will see that you simply drag the project into any existing project that depends on SwiftSyntax.

**Reply by Narsail** [reply]
That's what I'm doing currently but it had the downside that you have to do this a lot. Especially when Xcode is having its 5min in previews and you have to restart it again.

I was under the impression that if the library would directly integrate it, that would eliminate the need to move it into your project again and again.
